packages/abstractions/kiota_abstractions/serialization/parse_node.py

```diff
@@ -146,7 +146,7 @@ def get_collection_of_enum_values(self) -> List[K]:
         pass
 
     @abstractmethod
-    def get_enum_value(self) -> Enum:
+    def get_enum_value(self, enum_class: K) -> Optional[K]:
```

Comment:
leaving the comment here, but the argument for get object value factory should be generic so the return type is also generic